src/Storage/Sha1FileHashingService.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace SilverStripe\Assets\Storage;
+
+use InvalidArgumentException;
+use League\Flysystem\FileNotFoundException;
+use League\Flysystem\Filesystem;
+use League\Flysystem\Util;
+use Psr\SimpleCache\CacheInterface;
+use SilverStripe\Assets\Flysystem\FlysystemAssetStore;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Injector\Injector;
+
+/**
+ * Utility for computing and comparing unique file hash. All `$fs` parameters can either be:
+ * * an `AssetStore` constant VISIBILITY constant or
+ * * an actual `Filesystem` object.
+ *
+ * @internal This interface is not part of the official SilverStripe API and may be altered in minor releases.
+ */
+class Sha1FileHashingService implements FileHashingService
+{
+    use Configurable;
```

Comment:
Should this implement `Flushable`? The `flush()` method would need to be static in this case, which kind of doesn't work. Maybe we need a wrapper class that's using the injector service?

Reply:
Made FileHashingService flushable.